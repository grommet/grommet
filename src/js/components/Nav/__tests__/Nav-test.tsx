import React from 'react';
import 'jest-styled-components';
import 'jest-axe/extend-expect';
import 'regenerator-runtime/runtime';
import { hpe } from 'grommet-theme-hpe';

import { axe } from 'jest-axe';
import { render } from '@testing-library/react';

import { Grommet, Nav } from '../..';

describe('Nav', () => {
  test('should have no accessibility violations', async () => {
    const { container } = render(
      <Grommet theme={hpe}>
        <Nav />
      </Grommet>,
    );

    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container).toMatchSnapshot();
  });
  test('should render without grommet wrapper', async () => {
    const { container } = render(<Nav />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
