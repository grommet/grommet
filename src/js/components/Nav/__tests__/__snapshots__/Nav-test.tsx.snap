// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Nav should have no accessibility violations 1`] = `
.c1 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div>
  <div
    class="c0"
  >
    <nav
      class="c1"
    />
  </div>
</div>
`;

exports[`Nav should render without grommet wrapper 1`] = `
.c0 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

<nav
  class="c0"
/>
`;
